src/strategy/user-with-id-strategy.ts

```diff
@@ -0,0 +1,13 @@
+import { Strategy } from '../strategy';
+import { Context } from '../context';
+
+export class UserWithIdStrategy extends Strategy {
```

Comment:
Identical to `RemoteAddressStrategy`. Share code? 😀 

Or make it something more generic than `User` or `Address`